pallets/manta-pay/Cargo.toml

```diff
@@ -62,30 +58,33 @@ rand = { version = "0.8.4", default-features = false, optional = true }
+frame-support = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.16", default-features = false }
+frame-system = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.16", default-features = false }
+sp-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.16", default-features = false }
 sp-std = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.16", default-features = false }
 scale-codec = { package = "parity-scale-codec", version = "2.3.1", default-features = false, features = ["derive", "max-encoded-len"] }
 scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
 
 # manta dependencies
-manta-accounting = { git = "https://github.com/manta-network/manta-rs", default-features = false }
-manta-crypto = { git = "https://github.com/manta-network/manta-rs", default-features = false }
-manta-pay = { git = "https://github.com/manta-network/manta-rs", default-features = false, features = ["groth16", "scale"] }
-manta-sdk = { git = "https://github.com/manta-network/sdk", default-features = false }
-manta-util = { git = "https://github.com/manta-network/manta-rs", default-features = false }
+manta-accounting = { git = "https://github.com/manta-network/manta-rs.git", default-features = false }
+manta-crypto = { git = "https://github.com/manta-network/manta-rs.git", default-features = false }
+manta-pay = { git = "https://github.com/manta-network/manta-rs.git", default-features = false, features = ["groth16", "scale"] }
+manta-sdk = { git = "https://github.com/manta-network/sdk.git", default-features = false }
+manta-util = { git = "https://github.com/manta-network/manta-rs.git", default-features = false }
 manta-primitives = { path = "../../primitives", default-features = false}
 
 [dev-dependencies]
 bencher = "0.1.5"
 criterion = "0.3.4"
 lazy_static = "1.4.0"
-manta-accounting = { git = "https://github.com/manta-network/manta-rs", features = ["test"] }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.16"}
-sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.16"}
+manta-accounting = { git = "https://github.com/manta-network/manta-rs.git", features = ["test"] }
+sp-core = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.16"}
+sp-io = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.16"}
 pallet-balances = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.16" }
 pallet-assets = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.16" }
 pallet-asset-manager = { path = "../asset-manager"}
-
+manta-util = { git = "https://github.com/manta-network/manta-rs.git"}
+manta-sdk = { git = "https://github.com/manta-network/sdk.git", features = ["download"] }
+tempfile = "3.3.0"
```

Comment:
i don't see which change needs tempfile and rand, could you explain?

Reply:
I'm not that familiar with manta-pay, but I suppose these test cases needs them.

Reply:
Ok i saw both being used in testcode
/resolve